modules/infra/api/graph.c

```diff
@@ -37,15 +41,34 @@ static struct api_out graph_dump(const void * /*request*/, void **response) {
 	if ((ret = rte_graph_export(graph_name, stream)) < 0)
 		goto end;
 	fflush(stream);
-	resp_len = sizeof(*resp) + buf_len + 1;
+
+	if (!(req->flags & GR_INFRA_GRAPH_DUMP_F_ERRORS)) {
+		// Remove sink nodes from the output. They all have a "darkorange" color.
+		// Also remove non-sink nodes that contain "error" in their name.
+		char *line = buf;
+		char *eol;
+		while ((eol = strchr(line, '\n')) != NULL) {
+			*eol = '\0';
+			if (strstr(line, "darkorange") || strstr(line, "error")) {
+				// Skip the whole line.
+				memmove(line, eol + 1, buf_len - (eol + 1 - buf));
+				buf_len -= eol + 1 - line;
+			} else {
+				*eol = '\n'; // restore newline char
+				line = eol + 1;
+			}
+		}
+	}
+
+	resp_len = sizeof(*resp) + buf_len;
 	if ((resp = calloc(1, resp_len)) == NULL) {
 		ret = -ENOMEM;
 		resp_len = 0;
 		goto end;
 	}
 
-	resp->len = buf_len + 1;
```

Comment:
This change is confusing, as it is actually unrelated to the change.
This should go to a separate commit (along with the update in cli/graph.c lines 22/23 hunk).

Reply:
done